### PR TITLE
BUG: Handle all-NA blocks in concat

### PR DESCRIPTION
**Pull request by TomAugspurger**
Previously we special cased all-na blocks. We should only do that for
non-extension dtypes.

### CODE REVIEW

File: pandas/tests/extension/base/reshaping.py [hunk 1]
```diff
@@ -25,6 +25,21 @@ def test_concat(self, data, in_frame):
         assert dtype == data.dtype
         assert isinstance(result._data.blocks[0], ExtensionBlock)
 
+    @pytest.mark.parametrize('in_frame', [True])
```
Comment (jschendel) [inline pandas/tests/extension/base/reshaping.py]: Should this be `[True, False]`?  If `False` isn't supported yet, might be nice to write a quick comment, as separate code paths for `False` exist in the test.

Comment (TomAugspurger) [reply pandas/tests/extension/base/reshaping.py]: Yeah, was debugging that case. Thanks.

**Comment by jreback** [comment]
lgtm.

**Comment by TomAugspurger** [comment]
The linting error was fixed on master: `pandas/_libs/tslibs/period.pyx:1358:12: W291 trailing whitespace`

**Pull request merged**